drivers/clock_control/clock_control_nrf2_lfclk.c

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nordic_nrf_lfclk
+
+#include "clock_control_nrf2_common.h"
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/clock_control/nrf_clock_control.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(clock_control_nrf2, CONFIG_CLOCK_CONTROL_LOG_LEVEL);
+
+#include <nrfs_clock.h>
+
+BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 1,
+	     "multiple instances not supported");
+
+/* TODO: get from Kconfig */
+#define UPDATE_TIMEOUT K_MSEC(10)
+
+static const struct clock_options {
+	uint16_t accuracy : 15;
+	uint16_t precision : 1;
+	nrfs_clock_src_t src;
+} clock_options[] = {
+	/* TODO: use real accuracy values, handle other LFXO configurations */
+	{
+		.accuracy = 100,
+		.precision = 0,
+		.src = NRFS_CLOCK_SRC_LFCLK_XO_PIXO
+	},
+	{
+		.accuracy = 50,
```

Comment:
Please remember and be ready that some sources have the same accuracy. For instance.  Difference between NRFS_CLOCK_SRC_LFCLK_XO_PIERCE and NRFS_CLOCK_SRC_LFCLK_XO_PIERCE_HP is that  HP have had better precision. Accuracy and frequance is the same.